third_party/amd/lib/TritonAMDGPUToLLVM/LoadStoreOpToLLVM.cpp

```diff
@@ -720,24 +720,144 @@ bool supportsGlobalAtomicF16PackedAndDpp(triton::AMD::ISAFamily isaFamily) {
          isaFamily == triton::AMD::ISAFamily::CDNA3;
 }
 
-Value generateI32DppMove(PatternRewriter &rewriter, Value val, int dppCtrl) {
-  assert(val.getType().isInteger(32));
+Value generateIntDppMove(PatternRewriter &rewriter, Value val, int dppCtrl) {
+  Type movedTy = val.getType();
+  assert(movedTy.isInteger());
   auto loc = val.getLoc();
-  Value old = i32_val(0);
+  Value old = int_val(movedTy.getIntOrFloatBitWidth(), 0);
   int rowMask = 0b1111;  // enable all rows
   int bankMask = 0b1111; // enable all banks
   bool boundCtrl = false;
   auto dppMovOp = rewriter.create<ROCDL::DPPUpdateOp>(
-      loc, i32_ty, old, val, dppCtrl, rowMask, bankMask, boundCtrl);
+      loc, movedTy, old, val, dppCtrl, rowMask, bankMask, boundCtrl);
   return dppMovOp.getResult();
 }
 
-Value shiftLeftI32ByDpp(PatternRewriter &rewriter, Value val) {
-  return generateI32DppMove(rewriter, val, 0x101); // shift left 1 lane
+Value shiftLeftIntByDpp(PatternRewriter &rewriter, Value val, int step = 1) {
+  return generateIntDppMove(rewriter, val, 0x100 + step); // shift left
 }
 
-Value shiftRightI32ByDpp(PatternRewriter &rewriter, Value val) {
-  return generateI32DppMove(rewriter, val, 0x111); // shift right 1 lane
+Value shiftRightIntByDpp(PatternRewriter &rewriter, Value val, int step = 1) {
+  return generateIntDppMove(rewriter, val, 0x110 + step); // shift right
+}
+
+void generateBitonicSort(PatternRewriter &rewriter, Value key, Value val) {
+  // todo: extend logic to block size
```

Comment:
Can you add some comments to explain the algorithm and impl in high level?

Also please add inline comments for the following impl to explain what each step is doing given it's fairly compute heavy.